src/angular-modal-service.js

```diff
@@ -135,18 +135,22 @@ module.factory('ModalService', ['$animate', '$document', '$compile', '$controlle
           }
 
           function cleanUpClose(result) {
-
-            //  Resolve the 'close' promise.
-            closeDeferred && closeDeferred.resolve(result);
```

Comment:
You still need to resolve `closeDeferred`, just do it under the `if (!modalElement)`